src/main/java/com/faforever/api/data/checks/IsEntityOwner.java

```diff
@@ -1,30 +1,25 @@
 package com.faforever.api.data.checks;
 
 import com.faforever.api.data.domain.OwnableEntity;
+import com.faforever.api.security.ElideUser;
 import com.faforever.api.security.FafUserDetails;
-import com.yahoo.elide.security.ChangeSpec;
-import com.yahoo.elide.security.RequestScope;
-import com.yahoo.elide.security.checks.InlineCheck;
+import com.yahoo.elide.core.security.ChangeSpec;
+import com.yahoo.elide.core.security.RequestScope;
+import com.yahoo.elide.core.security.checks.OperationCheck;
 
 import java.util.Optional;
 
 public class IsEntityOwner {
 
   public static final String EXPRESSION = "is entity owner";
 
-  public static class Inline extends InlineCheck<OwnableEntity> {
+  public static class Inline extends OperationCheck<OwnableEntity> {
 
     @Override
     public boolean ok(OwnableEntity entity, RequestScope requestScope, Optional<ChangeSpec> changeSpec) {
-      Object opaqueUser = requestScope.getUser().getOpaqueUser();
-      return opaqueUser instanceof FafUserDetails
-        && entity.getEntityOwner() != null
-        && entity.getEntityOwner().getId() == ((FafUserDetails) opaqueUser).getId();
-    }
-
-    @Override
-    public boolean ok(com.yahoo.elide.security.User user) {
-      return false;
+      final ElideUser caller = (ElideUser) requestScope.getUser();
+      return entity.getEntityOwner() != null
+        && entity.getEntityOwner().getId().equals(caller.getFafUserDetails().map(FafUserDetails::getId).orElse(null));
```

Comment:
Maybe Object.equals instead? More readable, slightly more null safe.

Reply:
I don't want to null == null be true